crates/matrix-sdk/src/room_preview.rs

```diff
@@ -69,18 +77,27 @@ pub struct RoomPreview {
     ///
     /// Set to `None` if the room is unknown to the user.
     pub state: Option<RoomState>,
+
+    /// The `m.room.direct` state of the room, if known.
+    pub is_direct: Option<bool>,
 }
 
 impl RoomPreview {
     /// Constructs a [`RoomPreview`] from the associated room info.
     ///
     /// Note: not using the room info's state/count of joined members, because
     /// we can do better than that.
-    fn from_room_info(
+    async fn from_room_info(
+        client: &Client,
```

Comment:
nit: since both callers have a handle to the `room` from the Client, can you pass `is_direct: bool` here instead?

Reply:
Sure, but I'd still have to call `room.is_direct().await.unwrap_or_default()`, is it fine to assume it's not direct at this point?

Reply:
I'd still get an `Option<bool>` in some cases. Maybe I'll just pass the nullable `is_direct` field to the clients and let them decide what to do with it.

Reply:
I'll still need to pass a `Client` around since I need it to be able to call `RoomPreview::leave()`, as there's no other way to leave an invited/knocked room if we change those to return a `RoomPreview` instead. Should I still calculate the `is_direct` value outside, if the client can do it inside with the latest info?